mono/mini/helpers.c

```diff
@@ -263,7 +263,7 @@ mono_disassemble_code (MonoCompile *cfg, guint8 *code, int size, char *id)
 
 	fflush (stdout);
 
-#if defined(__arm__) || defined(__aarch64__)
+#if (defined(__arm__) || defined(__aarch64__)) && !defined(TARGET_OSX)
```

Comment:
This isn't necessary on apple silicon? Can you elaborate?

Reply:
It's not necessary on apple platforms, but previously this code was never ran on apple because it's only used on desktop.

Reply:
Ah okay, I see.